### ISSUE TITLE
Uncertainties + pint

### ISSUE THREAD

**Issue by SengerM** [lebigot/uncertainties#86]
I am trying to combine uncertainties and [pint](https://pint.readthedocs.io/en/latest/) to avoid tedious calculations. When operations consist only in +,-,* and / everything works perfectly. The problem arises when I try to use other functions of the uncertainties package such as cosine(), sqrt(), etc. A MWE of what I am trying to do is as follows

	import uncertainties as un
	from uncertainties.umath import *
	import pint
	ureg = pint.UnitRegistry()
	a = un.ufloat(10,1)*ureg.meter
	b = un.ufloat(2,0.1)*ureg.meter
	print(a/b) # This works nice
	print(sqrt(un.ufloat(3,1))) # This works nice
	print(sqrt(a/b)) # This fails

As far as I know the problem is that sqrt() is waiting for a ufloat object and I am passing a "pint type object". Is there any way to implement this?

**Comment by lebigot** [comment]
Indeed, Pint objects should be handled by Pint instead of the uncertainties module: units have to be handled somewhere. 

Now, how would you do the same if your Pint quantities didn't have uncertainties?

**Comment by SengerM** [comment]
For Pint quantities without uncertainties it can be done using Numpy:

	import pint
	import numpy
	ureg = pint.UnitRegistry()
	a = 10*ureg.meter
	b = 20*ureg.meter
	print(numpy.sqrt(a/b))

**Comment by lebigot** [comment]
And so what happens if you do the same with uncertainties?

**Comment by lebigot** [comment]
Thanks. I'll investigate some more the interaction between NumPy and non-basic types, then. My guess is that `uncertainties` could interface more with NumPy (in addition to what is already available in `uncertainties.unumpy`).
